### ISSUE TITLE
most-recent tasks by state for gscan

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc-flow#3976]
Cylc UI needs to display something similar to the old gscan tooltips:

![shot](https://user-images.githubusercontent.com/2362137/102441873-07788180-4088-11eb-9860-b10214680ec4.png)

Currently the datastore holds the state total numbers, but the "most recent" 5 (say) tasks-by-state needs to be sent from the back end too, so that gscan in the UI does not have to subscribe to and iterate over the whole task pool.

This is distinct from the n-window store.

**Comment by oliver-sanders** [comment]
Note1: The original requirement for this was to display the most recent failed tasks, however, with the global-update infrastructure at the time it was just as easy to do it for all states.

Note2: This was implemented in order to get the warning triangle working. This triangle popped up whenever tasks failed and could be dismissed in the client by clicking it.

Note3: GTK is horrible, I had to get the mouse x,y coordinates, then measure the squares and their border in order to work out which one you were hovering over. Web tech FTW.
